ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java

```diff
@@ -108,7 +108,16 @@ public final void check()
                 // TODO close window where first can change - maybe record nextDynamicCheckTime before probing first,
                 // then use compareAndSet? Although something will need to check before sleeping anyhow...
                 if (reaperElement == null) {
-                    nextDynamicCheckTime.set(Long.MAX_VALUE);
+                    if (!_inShutdown)
```

Comment:
@jmfinelli a crucial thing with this change is that this check of _inShutdown is in the `synchronized(this)` block and it might be worth adding a comment to that effect

Reply:
Thank you!